src/commands/registry.rs

```diff
@@ -90,3 +94,65 @@ impl Pull {
         Ok(())
     }
 }
+
+#[derive(Parser, Debug)]
+pub struct Login {
+    /// Username for the container registry
+    #[clap(long = "username", short = 'u')]
+    pub username: Option<String>,
+
+    /// Password for the container registry
+    #[clap(long = "password", short = 'p')]
+    pub password: Option<String>,
+
+    /// Take the password from stdin
+    #[clap(
+        long = "password-stdin",
+        takes_value = false,
+        conflicts_with = "password"
+    )]
+    pub password_stdin: bool,
+
+    #[clap()]
+    pub server: String,
+}
+
+impl Login {
+    pub async fn run(self) -> Result<()> {
+        let username = match self.username {
+            Some(u) => u,
+            None => {
+                println!("Username: ");
```

Comment:
Hmm, this will go to a new line and put the cursor at the start of that new line.  It would be nicer to do it on one line using `print!` and `flush`, or the `dialoguer` crate has a helper for this (`dialoguer::Input::new().with_prompt(...).interact_text()`).

(We use `dialoguer` in other commands including templates and plugins.)

Reply:
Ah, great idea. Updated.